main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreNotInstalledInfoBar.cs

```diff
@@ -26,70 +26,38 @@
 
 using MonoDevelop.Core;
 using MonoDevelop.Ide;
-using MonoDevelop.Ide.Gui;
-using System;
+using MonoDevelop.Ide.Gui.Components;
 
 namespace MonoDevelop.DotNetCore
 {
-	class DotNetCoreNotInstalledDialog : IDisposable
+	class DotNetCoreNotInstalledInfoBar
 	{
-		static readonly string defaultMessage = GettextCatalog.GetString (".NET Core SDK is not installed. This is required to build and run .NET Core projects.");
-
-		GenericMessage message;
-		AlertButton downloadButton;
 		string downloadUrl = DotNetCoreDownloadUrl.GetDotNetCoreDownloadUrl ();
 
-		public DotNetCoreNotInstalledDialog ()
-		{
-			Build ();
-		}
-
-		void Build ()
-		{
-			message = new GenericMessage {
-				Text = defaultMessage,
-				DefaultButton = 1,
-				Icon = Stock.Information
-			};
-
-			downloadButton = new AlertButton (GettextCatalog.GetString ("Download .NET Core..."));
-			message.Buttons.Add (AlertButton.Cancel);
-			message.Buttons.Add (downloadButton);
-
-			message.AlertButtonClicked += AlertButtonClicked;
-		}
-
-		void AlertButtonClicked (object sender, AlertButtonEventArgs e)
-		{
-			if (e.Button == downloadButton)
-				IdeServices.DesktopService.ShowUrl (downloadUrl);
-		}
-
-		public void Dispose ()
-		{
-			message.AlertButtonClicked -= AlertButtonClicked;
-		}
+		public string Message { get; set; }
+		public bool IsUnsupportedVersion { get; set; }
+		public bool IsNetStandard { get; set; }
+		public DotNetCoreVersion RequiredDotNetCoreVersion { get; set; }
+		public string CurrentDotNetCorePath { get; set; }
 
-		public void Show ()
+		public void Prompt ()
 		{
-			if (IsUnsupportedVersion || IsNetStandard) //for .net standard we'll show generic message
+			var items = new InfoBarItem [] {
+				new InfoBarItem (GettextCatalog.GetString ("Download .NET Core"), InfoBarItemKind.Hyperlink, DownloadButtonClicked, true)
```

Comment:
Use a button type here:
```suggestion
				new InfoBarItem (GettextCatalog.GetString ("Download .NET Core"), InfoBarItemKind.Button, DownloadButtonClicked, true)
```

Reply:
Idiomatically a button is an action and a hyperlink gives the user more information/context. "_Download .NET Core_" is definitely a call to "action", hence it should be a button (even tho it opens a browser with the download page, where the user can download the package).

Whereas something like "How to Install .NET Core" might be a hyperlink, indicating that the user can get more information/details about the problem identified by the infobar and optionally download .NET Core.

So if it really needs to be a hyperlink, rephrase the text.